### ISSUE TITLE
Optimization for boolean questions

### ISSUE THREAD

**Issue by rayslxu** [stanford-oval/thingtalk#367]
the `booleanExpression` in the boolean questions is currently not optimized. E..g, in the case when it's statically true/false, we could simply use `[true]`/`[false]` to avoid the computation, or even avoid running the query entirely.

**Comment by gcampax** [comment]
Yeah you'll want to add the filter optimization right away because it also normalizes the filters and sorts it by parameter name, which you need for training.
